src/libraries/Microsoft.PowerFx.Core/Public/Values/PagedRecordValue.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Public.Values
+{
+    // PagedRecordValue is a RecordValue that can be paged through.
+    // It is composed of the "nextLink" string and a table with the results
+    internal class PagedRecordValue : RecordValue
+    {
+        public RecordValue CurrentPage { get; private set; }
+
+        // Method used to retrieve the next page
+        public Func<FormulaValue> GetNextRecord { get; private set; }
+
+        public int MaxRows { get; private set; }
+
+        public PagedRecordValue(RecordValue recordValue, Func<FormulaValue> getNextRecord, int maxRows)
+            : base(recordValue.Type)
+        {
+            CurrentPage = recordValue;
+            GetNextRecord = getNextRecord;
+            MaxRows = maxRows;
+        }
+
+        protected internal override bool TryGetField(FormulaType fieldType, string fieldName, out FormulaValue result)
+        {
+            bool b = CurrentPage.TryGetField(fieldType, fieldName, out result);
+
+            // If we retrieve the table (via "value" field), we need to wrap it in a PagedTableValue
+            if (fieldType is TableType && result is TableValue tv)
```

Comment:
>if (fieldType is TableType && result is TableValue tv) [](http://example.com/codeflow?start=12&length=54)

can we use a more specific check here? like check on exact field name? Is it always called "value"?